### PR TITLE
Update kvproto in dependencies

### PR DESCRIPTION
**Pull request by HunDunDM**
### What problem does this PR solve?

* Update `kvproto` to develop `JointConsensus`

### What is changed and how it works?

* [x] Update `kvproto`: wait for [kvproto/pull/658](https://github.com/pingcap/kvproto/pull/658)
* Compatible with the change from `IsLearner` to `Role`.

### Check List

Tests

- Unit test

### CODE REVIEW

File: server/versioninfo/feature.go [hunk 1]
```diff
@@ -37,13 +36,22 @@ const (
 	// BatchSplit can speed up the region split.
 	// and PD will response the BatchSplit request.
 	BatchSplit
+	Version3_0
```
Comment (rleungx) [inline server/versioninfo/feature.go:39]: Why do we need 3_0/4_0?

Comment (HunDunDM) [reply server/versioninfo/feature.go]: This part of the code is split into another PR.

File: pkg/metautil/peer.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// limitations under the License.
+
+package metautil
+
+import (
+	"github.com/pingcap/kvproto/pkg/metapb"
+)
+
+// IsLearner judges whether the Peer's Role is Learner.
+func IsLearner(peer *metapb.Peer) bool {
```
Comment (rleungx) [inline pkg/metautil/peer.go:21]: Why do we need a `metautil` package, can we put it into `core/region.go`?

Comment (HunDunDM) [reply pkg/metautil/peer.go]: I'll put it into `core/meta.go`.

**Pull request merged**